environments/se/core/src/main/java/org/jboss/weld/environment/se/ContainerLifecycleObserver.java

```diff
@@ -0,0 +1,673 @@
+
+        private static final String OBSERVED_TYPE = "observedType";
+        private static final String CALLBACK = "callback";
+
+        @SuppressWarnings("checkstyle:magicnumber")
+        private static final int DEFAULT_PRIORITY = javax.interceptor.Interceptor.Priority.APPLICATION + 500;
+
+        private int priority = DEFAULT_PRIORITY;
+
+        private Type observedType;
+
+        private Collection<Class<? extends Annotation>> requiredAnnotations = Collections.emptySet();
+
+        private Builder(Type observedType) {
+            Preconditions.checkArgumentNotNull(observedType, OBSERVED_TYPE);
+            this.observedType = observedType;
+        }
+
+        /**
+         * Set the priority.
+         *
+         * @param priority
+         * @return self
+         */
+        public Builder<T> priority(int priority) {
+            this.priority = priority;
+            return this;
+        }
+
+        /**
+         * The annotations are only considered for {@link ProcessAnnotatedType}, i.e. they are ignored for other container lifecycle events.
+         *
+         * @param annotations
+         * @return self
+         * @see WithAnnotations
+         */
+        @SafeVarargs
+        public final Builder<T> withAnnotations(Class<? extends Annotation>... annotations) {
+            this.requiredAnnotations = new HashSet<>();
+            Collections.addAll(requiredAnnotations, annotations);
+            return this;
+        }
+
+        /**
+         * Set a callback used during observer notification. The first callback parameter is an event object and the second parameter is a {@link BeanManager}
+         * instance.
+         * <p>
+         * This is a terminal operation.
+         *
+         * @param callback
+         * @return the built observer
+         * @see ObserverMethod#notify(Object)
+         */
+        public ContainerLifecycleObserver<T> notify(BiConsumer<T, BeanManager> callback) {
+            Preconditions.checkArgumentNotNull(callback, CALLBACK);
+            return new ContainerLifecycleObserver<>(priority, observedType, callback, null, requiredAnnotations);
+        }
+
+        /**
+         * Set a callback used during observer notification. The callback parameter is an event object.
+         * <p>
+         * This is a terminal operation.
+         *
+         * @param callback
+         * @return the built observer
+         * @see ObserverMethod#notify(Object)
+         */
+        public ContainerLifecycleObserver<T> notify(Consumer<T> callback) {
+            Preconditions.checkArgumentNotNull(callback, CALLBACK);
+            return new ContainerLifecycleObserver<>(priority, observedType, null, callback, requiredAnnotations);
+        }
+
+    }
+
+    /**
+     * A synthetic extension, i.e. a collection of synthetic container lifecycle observers.
+     *
+     * @author Martin Kouba
+     */
+    public static class ContainerLifecycleObserverExtention implements SyntheticExtension {
```

Comment:
I think there is a typo here, 'ContainerLifecycleObserver**Extention**' should be 'ContainerLifecycleObserver**Extension**'.